operator/v1/types_ingress.go

```diff
@@ -225,6 +232,97 @@ var (
+	//
+	// If unset, the default is "Intermediate".
+	//
+	// +optional
+	Profile *SecurityProfileType `json:"profile,omitempty"`
+	// CustomSettings defines the schema for customizing the security settings of
+	// IngressController connections.
+	//
+	// +optional
+	CustomSettings *CustomProfileSettings `json:"customSettings,omitempty"`
+}
+
+// SecurityProfileType defines a security profile.
+type SecurityProfileType string
+
+const (
+	// SecurityProfileOldType maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Old_backward_compatibility
+	SecurityProfileOldType SecurityProfileType = "Old"
+	// SecurityProfileIntermediateType maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	SecurityProfileIntermediateType SecurityProfileType = "Intermediate"
+	// SecurityProfileModernType maps to:
+	// https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility
+	SecurityProfileModernType SecurityProfileType = "Modern"
+)
+
+// CustomProfileSettings defines the schema for a custom security profile.
+type CustomProfileSettings struct {
+	// Ciphers is used to specify the cipher algorithms that are negotiated
+	// during the SSL/TLS handshake with an IngressController. Each cipher must
+	// be an explicit, colon-delimited list of ciphers.
+	//
+	// If unset, the "Intermediate" ciphersuite [1] is used:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	Ciphers *string `json:"ciphers,omitempty"`
+	// SecurityProtocol is used to specify one or more encryption protocols
+	// that are negotiated during the SSL/TLS handshake with the IngressController.
+	//
+	// If unset, TLSv1.0, TLSv1.1, TLSv1.2 and TLSv1.3 are used.
+	//
+	// +optional
+	SecurityProtocol *SecurityProtocol `json:"securityProtocol,omitempty"`
+}
+
+// SecurityProtocol defines one or more security protocols used by
+// an IngressController to secure network connections.
+type SecurityProtocol struct {
+	// MinimumVersion enforces use of SecurityProtocolVersion or newer on
+	// SSL connections instantiated from an IngressController.
+	//
+	// If unset, TLSv1.0, TLSv1.1, TLSv1.2 and TLSv1.3 are used.
+	//
+	// +optional
+	MinimumVersion *SecurityProtocolVersion `json:"minimumVersion,omitempty"`
+	// MaximumVersion enforces use of SecurityProtocolVersion or older on
+	// SSL connections instantiated from an IngressController.
+	//
+	// If unset, TLSv1.0, TLSv1.1, TLSv1.2 and TLSv1.3 are used.
+	//
+	// +optional
+	MaximumVersion *SecurityProtocolVersion `json:"maximumVersion,omitempty"`
+}
+
+// SecurityProtocolVersion is a way to specify an IngressController
+// security protocol.
+type SecurityProtocolVersion string
+
+const (
+	// Versions are listed in order from oldest to newest.
+	//
+	// SecurityProtocolSSLv3Version is v3 of the SSL security protocol
+	SecurityProtocolSSLv3Version SecurityProtocolVersion = "SSLv3"
```

Comment:
We do not allow this in 3.11 and you can't enable it.  Let's not let people use it in 4.x.  It's decrepit, deprecated, and dangerous.

Reply:
I have removed `SecurityProtocolSSLv3Version`.